Added save and update functionality to app

Y0z64 opened the pull request:
- **Fixed UI and color issues**
- **Migrated db**
- **Added update and delete functions**
- **Added saving and updating**
- **Added manual save button to toolbar**
- **Made it, so the page tab title is the note title instead of the ID**


**Added auto-save functionality**: After every change to the markdown content it starts a countdown that is reset for every other change made during the countdown, finally, if it reaches 0 it will automatically save

Save button functions as an indicator that there is a save in progress, this indicator activates even if the save is initiated by the auto save functionality

## Issues:
- Tried to add a way to save when user changed route, but saving when the component unmounted caused a ddos attack to the rack API thanks to astro shenanigans
- Updating required deleting previous entry in rack and uploading a new one, it takes around ~13 to do the request which is not optimal and should be fixed

Y0z64 committed:
Fixed UI and colr issues

Y0z64 committed:
Migrated db

Y0z64 committed:
Added update and delete functions

Y0z64 committed:
Added saving and updating

Y0z64 committed:
Added manual save button

Y0z64 committed:
Merge branch 'main' into tomar-notas

Y0z64 committed:
Fixed UI issues

Gorchon reviewed (commented):
karamba

Y0z64 committed:
Functional delete button on note

DanielBrMz commented on src/components/react/MKEditor.tsx:
```diff
@@ -94,11 +129,11 @@ export default function App() {
-          className="outline-orange-400 outline w-full "
+          markdown={content}
+          className="w-full "
           contentEditableClassName="prose dark:text-white"
```
¿Es posible cambiar el color del texto que se renderiza cuando escribes títulos?

Por ejemplo: 
`# Titulo 1` -> 
# Titulo 1 

El Preview Text sigue apareciendo en color negro cuando el modo oscuro está activo

Y0z64 replied:
Yeah

Y0z64 replied:
Arregladisimo padre

DanielBrMz reviewed (approved):
Algo bonito, si puedes encontrar la manera de cambiar la configuración acerca del cometario que hice estaría atm

Y0z64 committed:
- **Added nearbyId to files table**
- **Added delete functionality to files, also deletes conversations related to the file"
- **Added button to delete files from dashboard"

Y0z64 committed:
Fixed UI issues that prevented text from showing in the right color when in darkmode

Y0z64 commented:
Mergeando sin aprovacion del @adpadillar como todo un chico peligroso que le gusta el riesgo
